src/getAttributes.js

```diff
@@ -1,4 +1,10 @@
-function attributeEntryToString([key, value]) {
+function attributeEntryToString(attribute, context) {
+  let [key, value] = attribute;
+
+  if (typeof value === "function") { // Callback must return a string or a number
+    value = value(context); // Run the provided callback and store the result
+  }
+
   if (typeof value !== "string" && typeof value !== "number")
     throw new Error(
       `Attribute "${key}" must have a value of type string or number not "${typeof value}".`
```

Comment:
I like the idea of adding support for callbacks with context. I think we should update the wording of the error message to reflect the allowance of callbacks by saying "…must have, or evaluate to, a value of…"

Reply:
That makes sense! Updating to the following:
```js
`Attribute "${key}" must have, or evaluate to, a value of type string or number, not "${typeof value}".`
```